astropy/io/fits/hdu/compressed/tests/test_compressed.py

```diff
@@ -735,7 +759,6 @@ def test_compressed_header_double_extname(self):
             hdul.writeto(self.temp("tmp.fits"), overwrite=True)
             with fits.open(self.temp("tmp.fits")) as hdul1:
                 hdu1 = hdul1[1]
-                assert len(hdu1._header._keyword_indices["EXTNAME"]) == 1
```

Comment:
Don't we need this test? It's meant to be confirming that there are not two EXTNAME headers in the output header. That was the purpose of the entire test (where we used to end up with a default EXTNAME and the one that was set externally).

Reply:
I'll double check!

Reply:
This was all from #11396 and was all a bit involved.